sigar_linux.go

```diff
@@ -343,12 +347,95 @@ func (self *NetIfaceList) Get() error {
 	return err
 }
 
+var inodeMatcher1 = regexp.MustCompile(`\Asocket:\[(\d+)\]\z`)
+var inodeMatcher2 = regexp.MustCompile(`\A\[0000\]:(\d+)\z`)
+
+func extractInode(linkName string) (uint64, error) {
+	results := inodeMatcher1.FindStringSubmatch(linkName)
+	// the inode number is held in the second result
+	if len(results) >= 2 {
+		inodeStr := results[1]
+		return strconv.ParseUint(inodeStr, 10, 64)
+	}
+	results = inodeMatcher2.FindStringSubmatch(linkName)
+	if len(results) >= 2 {
+		inodeStr := results[1]
+		return strconv.ParseUint(inodeStr, 10, 64)
+	}
+
+	return 0, errors.New("Failed to parse socket inode")
+}
+
+// Map pid to process name
+func buildPidMap(pids []int) map[int]string {
+	pidMap := make(map[int]string)
+	procState := ProcState{}
+	for _, pid := range pids {
+		err := procState.Get(pid)
+		if err == nil {
+			pidMap[pid] = procState.Name
+		}
+	}
+	return pidMap
+}
+
+func populatePidProcessName(netConnsPtr *[]NetConn) {
+	// Gather the list of pids
+	pids := ProcList{}
+	err := pids.Get()
+	if err != nil {
+		return
+	}
+
+	// For each process, read the links under the `fd` dir. If the linkName matches a specific form,
+	// the inode number is embedded in the name. See netstat source:
+	// https://github.com/ecki/net-tools/blob/master/netstat.c#L349
+	// Any discovered inodes are paired with the pid to match below
+	inodeCache := make(map[uint64]int)
+	for _, pid := range pids.List {
+		fdDir := procFileName(pid, "fd")
+		dir, err := os.Open(fdDir)
+		// Open the directory. Ignore any errors like lack of permission. We won't be able to read
+		// non-owned directories unless we're root, so much of the time this will fail.
```

Comment:
This comment mentions ignoring errors, but it seems like the code within is always checking `err == nil`.  Should this be checking that `err` is not equal to specific error types?

Reply:
We could check against the error type (e.g. using <a href="https://golang.org/pkg/os/#IsPermission">os.IsPermission</a>), but it doesn't seem like we could do anything different based on those errors. Did you have something in mind?

Reply:
Agreed, I don't think the behavior would change if you checked for that specific error.  I was just asking because the comment seemed to contradict the code

Reply:
Ah, I see! Let me try to clarify the comment.